### ISSUE TITLE
`--set` is ignored when `--reuse-values` is used during helm3 upgrade

### ISSUE THREAD

**Issue by SeMeKh** [helm/helm#6112]
Issuing a `helm3 upgrade --reuse-values --set "someKey=someValue"` command does not update the `someKey` value as expected.

Output of `helm version`:
```
version.BuildInfo{Version:"v3.0.0-alpha.2", GitCommit:"97e7461e41455e58d89b4d7d192fed5352001d44", GitTreeState:"clean", GoVersion:"go1.12.7"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:40:16Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): N/A

**Comment by karuppiah7890** [comment]
I'm able to reproduce the issue with this chart and setup:


1. Get this - [issue-6112-chart.zip](https://github.com/helm/helm/files/3452983/issue-6112-chart.zip)

```
$ unzip issue-6112-chart.zip
$ helm3 upgrade --install issue-6112 issue-6112-chart/ -f sample-values.yaml
$ # see configmap data
$ kubectl get configmap issue-6112-issue-6112-chart -o yaml
$ helm3 upgrade --install issue-6112 issue-6112-chart/ --reuse-values --set "someValue=newlySetValue"
$ # see configmap data again with no change
$ kubectl get configmap issue-6112-issue-6112-chart -o yaml
```

helm version:
```
version.BuildInfo{Version:"v3.0.0-alpha.2", GitCommit:"97e7461e41455e58d89b4d7d192fed5352001d44", GitTreeState:"clean", GoVersion:"go1.12.7"}
```


kubectl version:
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:32:14Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

Me and @nandhae will try to fix this bug and raise a PR 😄

**Comment by bacongobbler** [comment]
closed via #6134
